mycroft/client/speech/silence.py

```diff
@@ -0,0 +1,402 @@
+
+    skip_seconds: float = 0
+        Seconds of audio to skip before voice command detection starts
+
+    speech_seconds: float = 0.3
+        Seconds of speech before voice command has begun
+
+    before_seconds: float = 0.5
+        Seconds of audio to keep before voice command has begun
+
+    min_seconds: float = 1.0
+        Minimum length of voice command (seconds)
+
+    max_seconds: Optional[float] = 30.0
+        Maximum length of voice command before timeout (seconds, None for no timeout)
+
+    silence_seconds: float = 0.5
+        Seconds of silence before a voice command has finished
+
+    max_energy: Optional[float] = None
+        Maximum denoise energy value (None for dynamic setting from observed audio)
+
+    max_current_ratio_threshold: Optional[float] = None
+        Ratio of max/current energy below which audio is considered speech
+
+    current_energy_threshold: Optional[float] = None
+        Energy threshold above which audio is considered speech
+
+    silence_method: SilenceMethod = "vad_only"
+        Method for deciding if an audio chunk contains silence or speech
+    """
+
+    def __init__(
+        self,
+        sample_rate: int = 16000,
+        chunk_size: int = 960,
+        skip_seconds: float = 0,
+        min_seconds: float = 1,
+        max_seconds: typing.Optional[float] = 30,
+        speech_seconds: float = 0.3,
+        silence_seconds: float = 0.5,
+        before_seconds: float = 0.5,
+        max_energy: typing.Optional[float] = None,
+        max_current_ratio_threshold: typing.Optional[float] = None,
+        current_energy_threshold: typing.Optional[float] = None,
+        silence_method: SilenceMethod = SilenceMethod.VAD_ONLY,
+        plugin: str = None
+    ):
+        self.sample_rate = sample_rate
+        self.sample_width = 2  # 16-bit
+        self.sample_channels = 1  # mono
+        self.chunk_size = chunk_size
+        self.skip_seconds = skip_seconds
+        self.min_seconds = min_seconds
+        self.max_seconds = max_seconds
+        self.speech_seconds = speech_seconds
+        self.silence_seconds = silence_seconds
+        self.before_seconds = before_seconds
+
+        self.max_energy = max_energy
+        self.dynamic_max_energy = max_energy is None
+        self.max_current_ratio_threshold = max_current_ratio_threshold
+        self.current_energy_threshold = current_energy_threshold
+        self.silence_method = silence_method
+
+        # Verify settings
+        if self.silence_method in [
+            SilenceMethod.VAD_ONLY,
+            SilenceMethod.VAD_AND_RATIO,
+            SilenceMethod.VAD_AND_CURRENT,
+            SilenceMethod.ALL,
+        ]:
+            self.use_vad = True
+        else:
+            self.use_vad = False
+
+        if self.silence_method in [
+            SilenceMethod.VAD_AND_RATIO,
+            SilenceMethod.RATIO_ONLY,
+            SilenceMethod.ALL,
+        ]:
+            self.use_ratio = True
```

Comment:
I'd propose the following style change to condense this and raise a more specific exception (Applies to 161-171 too)
```python
self.use_ratio = self.silence_method in (...)
if self.use_ratio and not self.max_current_radio_threshold:
    raise RuntimeError(...)
```

Reply:
i dont want to do too many changes until this also becomes standard in mycroft to avoid future conflicts, let's keep it in sync as much as possible